src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationViewTaggerProvider.cs

```diff
@@ -92,15 +91,14 @@ protected override Task ProduceTagsAsync(TaggerContext<IClassificationTag> conte
             Debug.Assert(context.SpansToTag.IsSingle());
 
             var spanToTag = context.SpansToTag.Single();
-            var document = spanToTag.Document;
 
-            if (document == null)
+            _classificationService = _classificationService ?? 
+                spanToTag.Document?.Project.LanguageServices.GetService<IEditorClassificationService>();
```

Comment:
This is a bit subtle for my tastes - it isn't really clear that `Document` being null is an expected condition.